### DISCUSSION TITLE
What happened to the "remote follow" button?

### DISCUSSION THREAD

**Discussion by ncfavier** [mastodon/mastodon#21326] [Q&A] [answered]
Since mastodon 4, trying to follow someone on a different server brings up a popup that tells you to copy and paste a URL into your server's search box.

This seems a lot less useful than the old "remote follow" popup that just asked for your handle and redirected you. What happened to it?

**Comment by JustusW** [comment]
The fact you can't get that information natively from within the app is kinda mindboggling. It's a core requirement of a social network to be able to follow social connections. That's what social networks are _about_. In a way right now Mastodon is decentralized, but not a true social network...

**Comment by davidmh** [comment]
The feature was removed here: [`f7a1399` (#19319)](https://github.com/mastodon/mastodon/pull/19319/commits/f7a13998a24d5958a18a9907ad3a52f2eec63c6a#diff-9185eb9ca37cb2c8878e5006b357d404910be53f0089687b8fbb52be98296c61). 

With the following explanation in the description:

> Remove remote follow / remote interaction dialogs (superseded by in-app interaction modal)

Perhaps we can bring it back as an opt-in setting? If there's enough interest, I could work on it.

**Reply by NicolaeNMV** [reply]
I can see "Remote follow" in Pleorama and it's supper handy.
Why on earth is Mastodon not allowing this ? Copy & pasting ? That's really backward..

Following is indeed one of the most fundament things to do..

**Reply by Maffsie** [reply]
I was working on a patch to revert this, but unfortunately never got it to work quite right. Please reconsider its removal - the UX has been much worse since. :(

**Comment by renchap** [comment, marked as answer]
This is being worked on and we are trying to get it into the next release: https://github.com/mastodon/mastodon/pull/24842

**Reply by davidmh** [reply]
Looks like the update was just merged in a different PR #26075 🎉
